runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -1073,29 +1073,11 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 		J9IndexableObject *j9ArrayPtr = (J9IndexableObject *)arrayPtr;
 
 		if (InlineContiguous == getPreservedArrayLayout(forwardedHeader)) {
-			setDataAddrForContiguous(j9ArrayPtr);
-		} else {
-			setDataAddrForDiscontiguous(j9ArrayPtr, NULL);
-		}
-#endif /* J9VM_ENV_DATA64 */
-	}
-	/**
-	 * External fixup dataAddr API to update pointer of indexable objects.
-	 * Sets the dataAddr of either a contiguous or discomtiguous indexable
-	 * object.
-	 *
-	 * @param arrayPtr      Pointer to the indexable object whose size is required
-	 */
-	MMINLINE void
-	fixupDataAddr(omrobjectptr_t arrayPtr)
-	{
-#if defined(J9VM_ENV_DATA64)
-		J9IndexableObject *j9ArrayPtr = (J9IndexableObject *)arrayPtr;
-
-		if (InlineContiguous == getArrayLayout(j9ArrayPtr)) {
-			setDataAddrForContiguous(j9ArrayPtr);
+			if (shouldFixupDataAddrForContiguous((J9IndexableObject *)forwardedHeader->getObject())) {
+				setDataAddrForContiguous(j9ArrayPtr);
+			}
 		} else {
-			setDataAddrForDiscontiguous(j9ArrayPtr, NULL);
+			AssertDiscontiguousArrayDataNull(j9ArrayPtr);
```

Comment:
hm, wondering if this is fully correct....  dataAddr might be random where TLH is not batch cleared, and we are running in arraylets mode  and this happens to be 0-sized array allocated by JIT (it would be reasonable to not impose any initialization of dataAddr by JIT, in case of arraylets)

Reply:
From a discussion with @VermaSh today, it would be better if we did not impose dataAddr initialization to NULL, therefore, I will be removing this assert.